### PR TITLE
big merge for new workflow

### PR DESCRIPTION
**Pull request by daraghmartin**

### CODE REVIEW

File: options.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const optionsInputs = ["jiraUrl", "jiraFieldDefaults"];
+const optionsInputs = ["jiraUrl", "jiraFieldDefaults","jiraEpics", "apiKey", "apiToken"];
```
Comment (thachling) [inline options.js:1]: space

File: options.js [hunk 2]
```diff
@@ -19,10 +19,12 @@ async function loadInputs() {
           input.dataset.invalid = true;
         }
       }
-
       save(id, event.target.value);
     });
-  });
+
```
Comment (thachling) [inline options.js:24]: this hurts me

File: src/index.js [hunk 3]
```diff
@@ -9,28 +9,120 @@ function prependActionButton(button) {
...
+    select = select + (`<option value="${epics[key]}">${epics[key]} | ${key}</option>`)
+  }
+
+  select = select + '</select>]';
+
+  actionsList2.innerHTML = select + actionsList2.innerHTML;
+}
+
+function httpGet(theUrl)
+{
```
Comment (thachling) [inline src/index.js:28]: can we make this on the same line to keep it consistent with the other ones

File: src/index.js [hunk 4]
```diff
@@ -9,28 +9,120 @@ function prependActionButton(button) {
...
+    const xmlHttp = new XMLHttpRequest();
+    xmlHttp.open( "GET", theUrl, false ); // false for synchronous request
```
Comment (thachling) [inline src/index.js:30]: can we remove the spaces at start and end to keep it consistent with the other ones

File: src/index.js [hunk 5]
```diff
@@ -9,28 +9,120 @@ function prependActionButton(button) {
...
+    xmlHttp.send( null );
+    return xmlHttp.responseText;
+}
+
+function getCardMatches (url) {
```
Comment (thachling) [inline src/index.js:35]: wouldn't this be better returned as an object?

File: options.js (hunk 2 above)
Comment (TomLingham) [reply options.js:24]: Stop hurting @thachling, @daraghmartin

File: src/index.js [hunk 6]
```diff
@@ -9,28 +9,120 @@ function prependActionButton(button) {
...
+          const matches = getCardMatches(path);
+          const longPath = `https://trello.com${path}`
+
+          const cardID = matches[1];
+          const cardFullName = matches[2];
+
+          const url = `https://trello.com/1/cards/${cardID}`;
+          const card = JSON.parse(httpGet(url));
+
+          var cardDesc = card.desc;
```
Comment (thachling) [inline src/index.js:88]: can simplify this to
`const cardDesc = card.desc || card.name`

File: src/index.js [hunk 7]
```diff
@@ -9,28 +9,120 @@ function prependActionButton(button) {
...
+          if (!cardDesc) {
+            cardDesc = card.name;
+          }
 
-          const description = document.querySelector(
-            ".js-card-desc.current.markeddown"
-          ).innerHTML;
+          const cardShortName = card.name;
+          var jiraDescription = cardDesc;
```
Comment (thachling) [inline src/index.js:94]: use `let` instead of `var`

File: src/index.js (hunk 5 above)
Comment (TomLingham) [reply src/index.js:35]: Use fetch api
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

**Comment by TomLingham** [comment]
If you setup a prettier script that will fix all @thachling's style issues

File: src/jira.js [hunk 8]
```diff
@@ -3,6 +3,8 @@ async function api(path, options = {}, base = "api/2") {
 
   const url = `${settings.jiraUrl}/rest/${base}/${path}`;
 
+  console.log(url);
```
Comment (thachling) [inline src/jira.js:6]: did you want to leave this in?

File: src/jira.js [hunk 9]
```diff
@@ -46,7 +48,37 @@ export async function createIssue(summary, description) {
...
       )
-    });
+    };
+
+    const issue = await post("issue", data);
+
+    if (epic) {
+      try{
+        var data2 = {
+              "type": {
```
Comment (thachling) [inline src/jira.js:58]: you don't me quotation marks around the keys here

File: src/jira.js [hunk 10]
```diff
@@ -46,7 +48,37 @@ export async function createIssue(summary, description) {
...
+              "outwardIssue": {
+                  "key": epic
+              },
+              "comment": {
+                  "body": "Linked related issue!"
+              }
+          };
+
+
+        var x = await post("issueLink", data2);
```
Comment (thachling) [inline src/jira.js:73]: if you're not doing anything with x, you can just have
`await post('issueLink', data2);`

**Review by thachling — APPROVED** [review]
just some minor things around formatting which might not matter

**Comment by thachling** [comment]
i didn't realise it'd do an entry for each change :(

**Pull request closed without merging**